docs/mypy.md

```diff
@@ -0,0 +1,30 @@
+# mypy static checker
+
+Zulip now uses mypy. mypy is a compile-time type checker for python.
+For information about what mypy is, visit [http://mypy-lang.org/](http://mypy-lang.org/).
+
+mypy's type checking has been added to Zulip's Travis CI checks.
+
+## Running mypy on Zulip's code
+
+To run mypy on Zulip's code, you'll need to make sure that mypy is installed.
+mypy only runs on python 3. Zulip doesn't use the latest version of mypy available on pypi.
+It uses a specific commit from [mypy's git repository](https://github.com/python/mypy).
+
+*   If you installed Zulip's development environment using vagrant or provision.py,
+    mypy is already installed. To switch to an appropriate python 3 virtual environment, run
+
+        source /srv/zulip-py3-venv/bin/activate
```

Comment:
Can we just do the venv activation inside `tools/run-mypy` so the user doesn't need to know about it?